docs/explanation/locks.rst

```diff
@@ -0,0 +1,235 @@
+Linux kernel locks
+==================
+
+.. note::
```

Comment:
Should this be a `warning` or either a `tip` instead?

Reply:
I think a `note` is alright, referring to the specific kernel version used to write the documentation. 

However, I don't think we should be documenting kernel space interfaces. Are the documented locks and functions available from the user space and relevant to this RT Ubuntu's readers?

Reply:
It's an explanation, the user who end up reading this is willing to understand details of how it works, so from the point of view of what an **explanation** is supposed to be this is relevant.

Also, we already documented kernel space interfaces, for example, the schedulers one.